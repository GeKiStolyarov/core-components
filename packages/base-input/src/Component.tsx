/* eslint-disable react/jsx-props-no-spreading */

/**
 * Vendor
 */

import React, { useRef, useState } from 'react';
import cn from 'classnames';

/**
 * Styles
 */

import styles from './Component.module.css';

/**
 * Types
 */

export type BaseInputProps = {
    /** Размер компонента */
    size?: 's' | 'm' | 'l';
    /** Класс компонента */
    className?: string;
    /** Класс компонента */
    innerClassName?: string;
    /** Класс компонента */
    inputClassName?: string;
    /** Значение поля */
    value?: string;
    /** Плейсхолдер */
    placeholder?: string;
    /** Лейбл компонента */
    label?: React.ReactNode;
    /** Атрибут disabled */
    disabled?: boolean;
    /** Атрибут required */
    required?: boolean;
    /** Слот слева от инпута */
    leftAddons?: React.ReactNode;
    /** Слот справа от инпута */
    rightAddons?: React.ReactNode;
    /** Слот для отображения контента снизу */
    children?: React.ReactNode;
    /** Обработчик фокуса инпута */
    onFocus?: (event?: React.FocusEvent<HTMLInputElement>) => void;
    /** Обработчик блюра инпута */
    onBlur?: (event?: React.FocusEvent<HTMLInputElement>) => void;
    /** */
    onChange?: (event?: React.ChangeEvent<HTMLInputElement>) => void;
    /** Id компонента для тестов */
    dataTestId?: string;
};

/**
 * Expo
 */

export const BaseInput: React.FC<BaseInputProps> = ({
    size='s',
    className,
    innerClassName,
    inputClassName,
    value,
    disabled,
    required,
    placeholder,
    label,
    leftAddons,
    rightAddons,
    children,
    onFocus,
    onBlur,
    onChange,
    dataTestId,
    ...restProps
}) => {
    const inputRef = useRef(null);
    const [focused, setFocused] = useState(false);
    const [filled, setFilled] = useState(!!value);

    const hasLabel = !!label;

    function handleInputFocus(e: React.FocusEvent<HTMLInputElement>) {
        setFocused(true);

        if (onFocus) {
            onFocus(e);
        }
    }

    function handleInputBlur(e: React.FocusEvent<HTMLInputElement>) {
        setFocused(false);

        if (onBlur) {
            onBlur(e);
        }
    }

    function handleInputChange(e: React.ChangeEvent<HTMLInputElement>) {
        setFilled(!!e.target.value);

        if (onChange) {
            onChange(e);
        }
    }

    return (
        <div
            className={ cn(
                styles.component,
                {
                    [styles.focused]: focused,
                    [styles.disabled]: disabled,
                    [styles.filled]: filled,
                    [styles.hasLabel]: hasLabel
                },
                className
            ) }
        >
            <div className={ cn(styles.inner, innerClassName) }>
                { label && (
                    <div className={ cn(styles.label) }>
                        { label }
                    </div>
                ) }

                { leftAddons && (
                    <div className={ cn(styles.leftAddons) }>
                        { leftAddons }
                    </div>
                ) }

                <input
                    className={ cn(styles.input, styles[size], inputClassName) }
                    ref={ inputRef }
                    value={ value }
                    placeholder={ placeholder }
                    disabled={ disabled }
                    required={ required }
                    onChange={ handleInputChange }
                    onFocus={ handleInputFocus }
                    onBlur={ handleInputBlur }
                    data-test-id={ dataTestId }
                    // REVIEW: Думаю в данном кейсе это нормальное решение
                    { ...restProps }
                />

                { rightAddons && (
                    <div className={ cn(styles.rightAddons) }>
                        { rightAddons }
                    </div>
                ) }
            </div>

            { children }
        </div>
    );
};
