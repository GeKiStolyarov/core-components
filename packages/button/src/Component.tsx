/**
 * Vendor
 */

import React from 'react';
import cn from 'classnames';

/**
 * Styles
 */

import styles from './Component.module.css';

/**
 * Types
 */

type Props = {
    type?: 'primary' | 'secondary' | 'outlined' | 'link' | 'ghost';
    title?: string;
    disabled?: boolean;
    htmlType?: 'button' | 'reset' | 'submit';
    href?: string;
    leftAddons?: React.ReactNode;
    rightAddons?: React.ReactNode;
    addonsClassName?: string;
    size?: 'xs' | 's' | 'm' | 'l';
    block?: boolean;
    className?: string;
    dataTestId?: string;
    onClick?: (e: React.MouseEvent) => void;
};

/**
 * Expo
 */

export const Button: React.FC<Props> = ({
    children,
    type = 'secondary',
    title,
    disabled = false,
    htmlType = 'button',
    href,
    leftAddons,
    rightAddons,
    addonsClassName,
    size = 'm',
    block = false,
    className = '',
    dataTestId,
    onClick,
}) => {
    const buttonProps = {
        title,
        className: cn(
            styles.component,
            styles[type],
            styles[size],
            {
                [styles.block]: block,
                [styles['icon-only']]: !children,
            },
            className,
        ),
        onClick,
        'data-test-id': dataTestId || null,
    };

    const buttonChildren = (
        <React.Fragment>
            {leftAddons && <span className={cn(styles.addons, addonsClassName)}>{leftAddons}</span>}
            {children && <span className={cn(styles.text)}>{children}</span>}
            {rightAddons && (
                <span className={cn(styles.addons, addonsClassName)}>{rightAddons}</span>
            )}
        </React.Fragment>
    );

    if (href) {
        return (
            <a {...buttonProps} href={href}>
                {buttonChildren}
            </a>
        );
    }

    return (
        // eslint-disable-next-line react/button-has-type
        <button {...buttonProps} type={htmlType} disabled={disabled}>
            {buttonChildren}
        </button>
    );
};
