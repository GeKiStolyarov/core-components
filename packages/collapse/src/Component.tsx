import React, { forwardRef, useCallback, useEffect, useRef, useState } from 'react';
import cn from 'classnames';
import { ArrowDownSBlackIcon } from '@alfalab/icons-classic/ArrowDownSBlackIcon';
import { ArrowUpSBlackIcon } from '@alfalab/icons-classic/ArrowUpSBlackIcon';
import { Link } from '@alfalab/core-components-link';
import debounce from 'lodash.debounce';
import styles from './index.module.css';

export type CollapseProps = {
    /**
     * Первоначальное состояние компонента
     *
     */
    expanded?: boolean;

    /**
     * Текст ссылки в `expanded` состоянии
     *
     */
    collapsedLabel?: string;

    /**
     * Текст ссылки в `collapsed` состоянии
     *
     */
    expandedLabel?: string;

    /**
     * Дочерние элементы `Collapse`
     */
    children?: React.ReactNode;

    /**
     * Дополнительный класс
     */
    className?: string;

    /**
     * Идентификатор компонента в DOM
     */
    id?: string;

    /**
     * Обработчик смены состояний `expanded/collapsed`
     */
    onExpandedChange?: (expanded?: boolean) => void;

    /**
     * Идентификатор для систем автоматизированного тестирования
     */
    dataTestId?: string;
};

export const Collapse = forwardRef<HTMLLabelElement, CollapseProps>((props: CollapseProps) => {
    const {
        expanded,
        collapsedLabel,
        expandedLabel,
        children,
        className,
        id,
        onExpandedChange,
        dataTestId,
    } = props;

    const contentRef = useRef<HTMLDivElement>(null);
    const contentCaseRef = useRef<HTMLDivElement>(null);
    const [isExpanded, setIsExpanded] = useState(expanded);

    const recalculate = useCallback(() => {
        let contentHeight;

        if (!contentCaseRef.current || !contentRef.current) {
            return;
        }

        if (isExpanded) {
            contentHeight = contentCaseRef.current.offsetHeight;
        } else {
            contentHeight = 0;
        }

        contentRef.current.style.height = `${contentHeight}px`;
    }, [isExpanded]);

    useEffect(() => {
        const handleResize = debounce(() => recalculate(), 300);

        window.addEventListener('resize', handleResize);

        return () => window.removeEventListener('resize', handleResize);
    }, [recalculate]);

    useEffect(() => recalculate(), [isExpanded, recalculate]);

    const contentClassName = cn(styles.content, {
        [styles.expandedContent]: isExpanded,
    });

    const ToggledIcon = isExpanded ? ArrowUpSBlackIcon : ArrowDownSBlackIcon;

    const handleExpandedChange = useCallback(() => {
        setIsExpanded(!isExpanded);
        if (onExpandedChange) onExpandedChange();
    }, [isExpanded, onExpandedChange]);

    return (
        <div className={cn(className, styles.collapse)} id={id} data-test-id={dataTestId}>
            <div ref={contentRef} className={contentClassName}>
                <div ref={contentCaseRef}>{children}</div>
            </div>
            <Link pseudo={true} onClick={handleExpandedChange}>
                {isExpanded ? expandedLabel : collapsedLabel}
                <ToggledIcon className={styles.icon} />
            </Link>
        </div>
    );
});
