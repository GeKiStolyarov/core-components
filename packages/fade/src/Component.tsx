import React, { ReactNode, forwardRef } from 'react';
import { Transition } from 'react-transition-group';
import { TransitionStatus } from 'react-transition-group/Transition';
import cn from 'classnames';

import {
    ComponentTransitionsProps,
    createTimeout,
    createTransitionStyles,
    reflow,
    useForkRef,
} from './utils';

import styles from './index.module.css';

export type FadeProps = {
    /**
     * Управление видимостью элемента
     */
    show: boolean;

    /**
     * Дополнительный класс
     */
    className?: string;

    /**
     * Управление поведением фокуса
     */
    tabIndex?: number;

    /**
     * Контент
     */
    children?: ReactNode;

    /**
     * Идентификатор для систем автоматизированного тестирования
     */
    dataTestId?: string;
} & ComponentTransitionsProps;

/**
 * Заимствовано из [MUI](https://material-ui.com/)
 */
export const Fade = forwardRef<Element, FadeProps>((props, ref) => {
    const {
        appear = false,
        children,
        className,
        dataTestId,
        delayTimeout = 0,
        show,
        onEnter,
        onExit,
        onExited,
        timeout = 500,
        tabIndex,
    } = props;

    const nodeRef = React.useRef(null);
    const foreignRef = useForkRef((children as { ref: React.Ref<typeof children> }).ref, ref);
    const handleRef = useForkRef(nodeRef, foreignRef);

    const handleEnter = (node: HTMLElement, isAppearing: boolean) => {
        const element = node;
        reflow(element); // Для гарантии воспроизведения анимации с ее начала.

        const { duration, delay } = createTransitionStyles({ timeout, delayTimeout }, 'enter');

        element.style.transitionDuration = duration;
        element.style.transitionDelay = delay;

        if (onEnter) {
            onEnter(element, isAppearing);
        }
    };

    const handleExit = (node: HTMLElement) => {
        const element = node;
        const { duration, delay } = createTransitionStyles({ timeout, delayTimeout }, 'exit');

        element.style.transitionDuration = duration;
        element.style.transitionDelay = delay;

        if (onExit) {
            onExit(element);
        }
    };

    const handleExited = (node: HTMLElement) => {
        if (onExited) {
            onExited(node);
        }
    };

    return (
        <Transition
            appear={appear}
            in={show}
            timeout={createTimeout({ timeout, delayTimeout })}
            onEnter={handleEnter}
            onExit={handleExit}
            onExited={handleExited}
        >
            { (status: string) => (
                <div
                    className={cn(
                        styles.fade,
                        { [styles.hidden]: status === 'exited' && !show },
                        (styles as { [status: string]: TransitionStatus })[status],
                        className,
                    )}
                    data-test-id={dataTestId}
                    tabIndex={tabIndex}
                >
                    {
                        React.isValidElement(children)
                        ? React.cloneElement(children, { ref: handleRef })
                        : children
                    }
                </div>
            ) }
        </Transition>
    );
});
