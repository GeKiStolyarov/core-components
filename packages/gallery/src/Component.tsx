import React from 'react';

// import { ControlBar, NavigationBar } from './components';

import styles from './index.module.css';

export const Gallery: React.FunctionComponent<any> = () => {
    return (
        // Подумать над Base<BaseModal>
        <div className={styles.container}>
            {/* <ControlBar filename='ContrBarControlBarControl.jsx' description='Изображение 1 из 8' /> */}
            {/* <NavigationBar /> */}
        </div>
    );
};
