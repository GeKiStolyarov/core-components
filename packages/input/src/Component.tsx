/**
 * Vendor
 */

import React, { useState } from 'react';
import cn from 'classnames';

/**
 * Styles
 */

import styles from './Component.module.css';

/**
 * Types
 */

export type InputProps = {
  /** Класс для контейнеров с аддонами */
  addonsClassName?: string;
  /** Флаг - растягивать инпута на ширину контейнера */
  block?: boolean;
  /** Слот для отображения контента снизу */
  children?: React.ReactNode;
  /** Класс компонента (контейнера) */
  className?: string;
  /** Размер компонента */
  size?: 's' | 'm' | 'l';
  /** Id компонента для тестов */
  dataTestId?: string;
  /** Значение по умолчанию */
  defaultValue?: string;
  /** Атрибут disabled */
  disabled?: boolean;
  /** Текст ошибки */
  error?: string;
  /** Текст подсказки */
  hint?: string;
  /** Атрибут type */
  htmlType?: 'number' | 'card' | 'email' | 'money' | 'password' | 'tel' | 'text';
  /** Класс инпута */
  inputClassName?: string;
  /** Лейбл компонента */
  label?: React.ReactNode;
  /** Слот слева от инпута */
  leftAddons?: React.ReactNode;
  /** Максимальное число символов */
  maxLength?: number;
  /** Обработчик блюра инпута */
  onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
  /** Обработчик ввода */
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
  /** Обработчик фокуса инпута */
  onFocus?: (event: React.FocusEvent<HTMLInputElement>) => void;
  /** Обработчик события нажатия на клавишу клавиатуры в момент, когда фокус находится на компоненте */
  onKeyDown?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
  /** Обработчик события отжатия на клавишу клавиатуры в момент, когда фокус находится на компоненте */
  onKeyUp?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
  /** Обработчик события вставки текста в поле */
  onPaste?: (event: React.ClipboardEvent<HTMLInputElement>) => void;
  /** Обработчик события прерывания касания по полю */
  onTouchCancel?: (event: React.TouchEvent<HTMLInputElement>) => void;
  /** Обработчик события прекращения касания по полю */
  onTouchEnd?: (event: React.TouchEvent<HTMLInputElement>) => void;
  /** Обработчик события перемещения при касании по полю */
  onTouchMove?: (event: React.TouchEvent<HTMLInputElement>) => void;
  /** Обработчик события касания по полю */
  onTouchStart?: (event: React.TouchEvent<HTMLInputElement>) => void;
  /** Плейсхолдер */
  placeholder?: string;
  /** Атрибут required */
  required?: boolean;
  /** Слот справа от инпута */
  rightAddons?: React.ReactNode;
  /** Последовательность перехода между контролами при нажатии на Tab */
  tabIndex?: number;
  /** Значение поля */
  value?: string;
};

/**
 * Expo
 */

// TODO: Этого не будет, когда появится компонент иконки.
const errorIcon = (
  <svg width='18' height='18' viewBox='0 0 18 18' fill='none' xmlns='http://www.w3.org/2000/svg'>
    <path d='M16.8709 14.432L9.80088 2.61199C9.71724 2.47327 9.59918 2.35852 9.45813 2.27886C9.31709 2.19921 9.15786 2.15735 8.99588 2.15735C8.8339 2.15735 8.67467 2.19921 8.53362 2.27886C8.39258 2.35852 8.27452 2.47327 8.19088 2.61199L1.13088 14.432C1.04895 14.5739 1.00552 14.7347 1.00489 14.8985C1.00426 15.0623 1.04647 15.2235 1.12731 15.366C1.20816 15.5085 1.32484 15.6273 1.46579 15.7108C1.60674 15.7943 1.76707 15.8395 1.93088 15.842H16.0609C16.2259 15.8426 16.3881 15.7993 16.5309 15.7165C16.6736 15.6337 16.7918 15.5145 16.8732 15.3709C16.9546 15.2274 16.9964 15.0648 16.9942 14.8998C16.9921 14.7348 16.9461 14.5733 16.8609 14.432H16.8709ZM9.87088 13.242C9.87088 13.4489 9.7887 13.6473 9.64242 13.7935C9.49614 13.9398 9.29775 14.022 9.09088 14.022H8.86088C8.75845 14.022 8.65702 14.0018 8.56239 13.9626C8.46775 13.9234 8.38177 13.866 8.30934 13.7935C8.23691 13.7211 8.17945 13.6351 8.14025 13.5405C8.10105 13.4458 8.08088 13.3444 8.08088 13.242V13.002C8.08088 12.572 8.43088 12.222 8.86088 12.222H9.13088C9.56188 12.222 9.91088 12.572 9.91088 13.002L9.87088 13.242ZM9.64088 11.242H8.31088L8.09088 5.80199H9.91088L9.64088 11.242Z' fill='#EF3124' />
  </svg>
);

export const Input = React.forwardRef<HTMLInputElement, InputProps>(({
  size='s',
  htmlType='text',
  block=false,
  addonsClassName,
  children,
  className,
  dataTestId,
  defaultValue,
  disabled,
  error,
  hint,
  inputClassName,
  label,
  leftAddons,
  maxLength,
  onChange,
  onFocus,
  onKeyDown,
  onKeyUp,
  onPaste,
  onTouchCancel,
  onTouchEnd,
  onTouchMove,
  onTouchStart,
  onBlur,
  placeholder,
  required,
  rightAddons,
  tabIndex,
  value
}, ref) => {
  const [focused, setFocused] = useState(false);

  const handleInputFocus = (e: React.FocusEvent<HTMLInputElement>) => {
    setFocused(true);

    if (onFocus) {
      onFocus(e);
    }
  };

  const handleInputBlur = (e: React.FocusEvent<HTMLInputElement>) => {
    setFocused(false);

    if (onBlur) {
      onBlur(e);
    }
  };

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    if (onChange) {
      onChange(e);
    }
  };

  const handleInputKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (onKeyDown) {
      onKeyDown(e);
    }
  };

  const handleInputKeyUp = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (onKeyUp) {
      onKeyUp(e);
    }
  };

  const handleInputPaste = (e: React.ClipboardEvent<HTMLInputElement>) => {
    if (onPaste) {
      onPaste(e);
    }
  };

  const handleInputTouchCancel = (e: React.TouchEvent<HTMLInputElement>) => {
    if (onTouchCancel) {
      onTouchCancel(e);
    }
  };

  const handleInputTouchEnd = (e: React.TouchEvent<HTMLInputElement>) => {
    if (onTouchEnd) {
      onTouchEnd(e);
    }
  };

  const handleInputTouchMove = (e: React.TouchEvent<HTMLInputElement>) => {
    if (onTouchMove) {
      onTouchMove(e);
    }
  };

  const handleInputTouchStart = (e: React.TouchEvent<HTMLInputElement>) => {
    if (onTouchStart) {
      onTouchStart(e);
    }
  };

  const rightAddonsRenderer = () => (
    <div className={ cn(styles.addons, addonsClassName) }>
      { error && errorIcon }
      { rightAddons }
    </div>
  );

  const leftAddonsRenderer = () => leftAddons && (
    <div className={ styles.addons }>
      { leftAddons }
    </div>
  );

  return (
    <div
      className={ cn(
        styles.component,
        {
          [styles.focused]: focused,
          [styles.disabled]: disabled,
          [styles.filled]: value,
          [styles.hasLabel]: label,
          [styles.hasError]: error,
          [styles.block]: block
        },
        className
      ) }
    >
      <div className={ styles.inner }>
        { leftAddonsRenderer() }

        <div className={ styles.inputWrapper }>
          { label && (
            <div className={ styles.label }>
              { label }
            </div>
          ) }

          <input
            className={ cn(styles.input, styles[size], inputClassName) }
            defaultValue={ defaultValue }
            disabled={ disabled }
            maxLength={ maxLength }
            onBlur={ handleInputBlur }
            onChange={ handleInputChange }
            onFocus={ handleInputFocus }
            onKeyDown={ handleInputKeyDown }
            onKeyUp={ handleInputKeyUp }
            onPaste={ handleInputPaste }
            onTouchCancel={ handleInputTouchCancel }
            onTouchEnd={ handleInputTouchEnd }
            onTouchMove={ handleInputTouchMove }
            onTouchStart={ handleInputTouchStart }
            placeholder={ placeholder }
            ref={ ref }
            required={ required }
            tabIndex={ tabIndex }
            type={ htmlType }
            value={ value }
            data-test-id={ dataTestId }
          />
        </div>

        { rightAddonsRenderer() }
      </div>

      { (error || hint) && (
        <span className={ styles.sub }>
          { error || hint }
        </span>
      ) }

      { children }
    </div>
  );
});
