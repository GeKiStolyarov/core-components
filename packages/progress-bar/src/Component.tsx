import React, { useMemo } from 'react';
import cn from 'classnames';

import styles from './index.module.css';

export type ProgressBarProps = {
    /** Css-класс для стилизации */
    className?: string;
    /** Значение заполненной части 0-100 */
    value?: number;
    /** Цвет заполнения */
    view?: 'positive' | 'negative';
    /** Id компонента для тестов */
    dataTestId?: string;
};

export const ProgressBar: React.FC<ProgressBarProps> = ({
    className,
    value = 0,
    view = 'positive',
    dataTestId,
}) => {
    const fillTranslate = useMemo(() => {
        return Number((-1 * (100 - value)).toFixed(2));
    }, [value]);

    return (
        <div
            role='progressbar'
            aria-valuenow={value}
            aria-valuemin={0}
            aria-valuemax={100}
            className={cn(styles.container, className)}
            data-test-id={dataTestId}
        >
            <div
                className={cn(styles.filled, styles[view])}
                style={{ transform: `translateX(${fillTranslate}%)` }}
            />
        </div>
    );
};
