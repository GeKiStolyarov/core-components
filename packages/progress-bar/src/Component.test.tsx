import React from 'react';
import { render } from '@testing-library/react';

import { ProgressBar } from './Component';

describe('ProgressBar', () => {
    describe('Snapshots tests', () => {
        it('should match snapshot', () => {
            expect(render(<ProgressBar />)).toMatchSnapshot();
        });

        it('should match with value snapshot', () => {
            expect(render(<ProgressBar value={32} />)).toMatchSnapshot();
        });

        it('should match with view snapshot', () => {
            expect(render(<ProgressBar view='negative' />)).toMatchSnapshot();
        });
    });

    describe('Attributes tests', () => {
        it('should set `data-test-id` attribute', () => {
            const testId = 'test-id';
            const { getByTestId } = render(<ProgressBar dataTestId={testId} />);

            expect(getByTestId(testId)).toHaveAttribute('data-test-id', testId);
        });
    });

    it('should use passed `value`', () => {
        const value = 72.45;
        const { container } = render(<ProgressBar value={value} />);

        expect(container.querySelector('.filled')).toHaveStyle('transform: translateX(-27.55%)');
    });

    describe('Classes tests', () => {
        it('should set `className` class to root', () => {
            const className = 'test-class';
            const { container } = render(<ProgressBar className={className} />);

            expect(container.firstElementChild).toHaveClass(className);
        });

        it('should set default color class', () => {
            const { container } = render(<ProgressBar />);

            expect(container.querySelector('.filled')).toHaveClass('positive');
        });

        it('should set color class', () => {
            const { container } = render(<ProgressBar view='negative' />);

            expect(container.querySelector('.filled')).toHaveClass('negative');
        });
    });

    it('should unmount without errors', () => {
        const { unmount } = render(<ProgressBar />);

        expect(unmount).not.toThrowError();
    });
});
