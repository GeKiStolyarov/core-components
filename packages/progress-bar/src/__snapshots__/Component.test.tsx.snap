// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Snapshots tests should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="container"
        role="progressbar"
      >
        <div
          class="filled positive"
          style="transform: translateX(-100%);"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="0"
      class="container"
      role="progressbar"
    >
      <div
        class="filled positive"
        style="transform: translateX(-100%);"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ProgressBar Snapshots tests should match with value snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="32"
        class="container"
        role="progressbar"
      >
        <div
          class="filled positive"
          style="transform: translateX(-68%);"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="32"
      class="container"
      role="progressbar"
    >
      <div
        class="filled positive"
        style="transform: translateX(-68%);"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ProgressBar Snapshots tests should match with view snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="container"
        role="progressbar"
      >
        <div
          class="filled negative"
          style="transform: translateX(-100%);"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="0"
      class="container"
      role="progressbar"
    >
      <div
        class="filled negative"
        style="transform: translateX(-100%);"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
